src/backend/services/CreditService.ts

```
@@ -0,0 +1,302 @@
+  async getBalance(userId: string): Promise<number> {
+    const cacheKey = `credits:${userId}`;
+    const balance = await this.redis.get(cacheKey);
+    
+    if (balance === null) {
+      // Fetch from database if not in cache
+      const user = await this.prisma.user.findUnique({
+        where: { id: userId },
+        select: { credits: true }
+      });
+      
+      balance = user?.credits?.toString() || '0';
```

Biome (1.9.4)

[error] Can't assign balance because it's a constant

This is where the variable is defined as constant

Unsafe fix: Replace const with let if you assign it to a new value.


(lint/correctness/noConstAssign)